core/bindings.rs

```diff
@@ -438,7 +464,41 @@ fn opcall_async<'s>(
   }
 }
 
-fn set_macrotask_callback(
+fn has_tick_scheduled(
+  scope: &mut v8::HandleScope,
+  _args: v8::FunctionCallbackArguments,
+  mut rv: v8::ReturnValue,
+) {
+  let state_rc = JsRuntime::state(scope);
+  let state = state_rc.borrow();
+  rv.set(to_v8(scope, state.has_tick_scheduled).unwrap());
+}
```

Comment:
Node does this by means of an ArrayBuffer, to avoid (slow) JS -> C++ calls. Not a blocker, merely an observation.

Reply:
Agreed, let's address in a follow up

Reply:
Actually I'm confused here - it seems to me that it's done by having a `TickInfo` bound from C++ to JS. Can you clarify what you meant?

Reply:
Node applies a trick where it makes a C struct available to JS land by using the struct's memory as a backing store for a typed array. That way C++ and JS land can communicate without expensive JS <-> C++ calls, they just read and write the struct's fields.

Reply:
I see, I'll be happy to explore this path, let's discuss it offline in details and I'll raise another PR